api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -36,15 +54,113 @@
+     * Get the cohort review info. If a cohort review does not exist, a default review will be created without
+     * any participants. The {@link CohortReviewInfo#getReviewCount()} will equal zero (0) and be the indicatior
+     * that the cohort review has not been fully created. If a cohort review does exist, the
+     * {@link CohortReviewInfo#getReviewCount()} will be > 0. If a cohort definition for the specified cohortId
+     * doesn't exist a {@link BadRequestException} will be thrown.
      *
      * @param cohortId
      * @param cdrVersionId
+     * @return
+     */
+    @Override
+    public ResponseEntity<CohortReviewInfo> getCohortReviewInfo(Long cohortId, Long cdrVersionId) {
+
+        CohortReview cohortReview = cohortReviewDao.findCohortReviewByCohortIdAndCdrVersionId(cohortId, cdrVersionId);
+        if (cohortReview == null) {
+            CohortDefinition definition = cohortDao.findCohortByCohortId(cohortId);
+            if (definition == null) {
+                throw new BadRequestException("Invalid Request: No Cohort definition matching cohortId: " + cohortId);
+            }
+            SearchRequest request = new Gson().fromJson(definition.getCriteria(), SearchRequest.class);
+            QueryResult result = bigQueryService.executeQuery(bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantCounterQuery(request)));
+            Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+            List<FieldValue> row = result.iterateAll().iterator().next();
+            long cohortCount = bigQueryService.getLong(row, rm.get("count"));
+
+            cohortReview = new CohortReview();
+            cohortReview.setCohortId(cohortId);
+            cohortReview.setCdrVersionId(cdrVersionId);
+            cohortReview.matchedParticipantCount(cohortCount);
+            cohortReview.setCreationTime(new Timestamp(System.currentTimeMillis()));
+            cohortReviewDao.save(cohortReview);
+        }
+
+        long count = participantCohortStatusDao.countByParticipantKey_CohortReviewId(cohortReview.getCohortReviewId());
+        CohortReviewInfo info = new CohortReviewInfo()
+                .cohortReviewId(cohortReview.getCohortReviewId())
+                .reviewCount(count)
+                .cohortCount(cohortReview.getMatchedParticipantCount());
+
+        return ResponseEntity.ok(info);
+    }
+
+    /**
+     * Create a cohort review per the specified cohortReviewId and size. If participant cohort status
+     * data exists for a review or no cohort review exists for cohortReviewId then throw a
+     * {@link BadRequestException}.
+     *
+     * @param cohortReviewId
+     * @param size
+     * @return
+     */
+    @Override
+    public ResponseEntity<CohortReviewInfo> createCohortReview(Long cohortReviewId, Integer size) {
+        if(participantCohortStatusDao.countByParticipantKey_CohortReviewId(cohortReviewId) > 0) {
```

Comment:
We probably want to throw BadRequestException if size is larger than some max value.